ui/src/components/pages/Pipeline/partials/Graph/index.tsx

```diff
@@ -176,9 +177,12 @@ const Flow = (props: FlowProps) => {
   const [statusPayload, setStatusPayload] = useState<any>(undefined);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [timerDateStamp, setTimerDateStamp] = useState<any>(undefined);
-  const [timer, setTimer] = useState<any>(undefined);
+  const timer = useRef<any>(undefined);
```

Comment:
I think the state you had before was fine. Would be type `number | undefined`.

Reply:
No, state was the actual problem, the state was changing before the react did batch updates. THat's the reason I went with ref.